internal/services/recommendation_poller.go

```diff
@@ -0,0 +1,126 @@
+package services
+
+import (
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/confluentinc/confluent-kafka-go/v2/kafka"
+	"github.com/go-playground/validator/v10"
+
+	"github.com/redhatinsights/ros-ocp-backend/internal/logging"
+	"github.com/redhatinsights/ros-ocp-backend/internal/model"
+	"github.com/redhatinsights/ros-ocp-backend/internal/types"
+	"github.com/redhatinsights/ros-ocp-backend/internal/utils"
+	"github.com/redhatinsights/ros-ocp-backend/internal/utils/kruize"
+)
+
+func FetchRecommendations(msg *kafka.Message, consumer_object *kafka.Consumer) {
+	log := logging.GetLogger()
+	validate := validator.New()
+	var kafkaMsg types.RecommendationKafkaMsg
+
+	if !json.Valid([]byte(msg.Value)) {
+		log.Errorf("received message on kafka topic is not vaild JSON: %s", msg.Value)
+		return
+	}
+	if err := json.Unmarshal(msg.Value, &kafkaMsg); err != nil {
+		log.Errorf("unable to decode kafka message: %s", msg.Value)
+		return
+	}
+	if err := validate.Struct(kafkaMsg); err != nil {
+		log.Errorf("invalid kafka message: %s", err)
+		return
+	}
```

Comment:
I can see this is similar in other processor as well, should this be moved to some common file?

Reply:
Yes, it should be moved. As discussed, we'll will look into the import cycle error w.r.t utils later